frappe/__init__.py

```diff
@@ -509,7 +509,9 @@ def sendmail(recipients=[], sender="", subject="No Subject", message="No Message
 whitelisted = []
 guest_methods = []
 xss_safe_methods = []
-def whitelist(allow_guest=False, xss_safe=False):
+allowed_http_methods_for_whitelisted_func = {}
+
+def whitelist(allow_guest=False, xss_safe=False, allow_http_methods=['GET', 'POST', 'PUT', 'DELETE']):
```

Comment:
better parameter name `methods` instead of `allow_http_methods`

Reply:
also this is not a good way to default python arguments.

```
def whitelist(allow_guest=False, xss_safe=False, methods=None):
    if not methods: methods = ['GET', 'POST', 'PUT', 'DELETE']
```

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments